### PR TITLE
feat: 🎸 Add documentation generator and metavar support

### PR DESCRIPTION
**Pull request by ccordi**

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+            version: $VERSION
+            nodeVersion: $NODE_VERSION
+            pid: $PID
+```
+
+<!-- EXAMPLE-END -->
+
+## Debugging Fabriclogger Configuration
+
+Fabriclogger has a dedicated flag `--print-config` to show the effective configuration after merging defaults, config file, environment variables and CLI flags.
```
Comment (ccordi) [inline docs/configuration.md:97]: This still needs to be added.

File: test/config/example1.fabriclogger.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
+fabric:
```
Comment (ziegfried) [inline test/config/example1.fabriclogger.yaml:1]: Maybe we can slim this example down a bit and make it more real-world vs populating every possible option.

File: config.schema.json [hunk 3]
```diff
@@ -0,0 +1,588 @@
+{
```
Comment (ziegfried) [inline config.schema.json:1]: I tried validating the example config against this schema and it currently fails:

```
$ npx ajv-cli validate -s config.schema.json -d test/config/example1.fabriclogger.yaml --strict-types=false
npx: installed 27 in 1.656s
test/config/example1.fabriclogger.yaml invalid
[ { keyword: 'type',
    dataPath: '/prometheus/ordererPort',
    schemaPath: '#/properties/ordererPort/type',
    params: { type: 'string' },
    message: 'should be string' } ]
```

**Review by ziegfried — CHANGES_REQUESTED** [review]
Looking good. One more thing we'll want to do is to update the root README to refer to the configuration documents (or sections within those). We can also probably remove some of the content in the root README that is now redundant or outdated, such as the environment variables table at the bottom and instead just link to the CLI reference section.

File: test/config/example1.fabriclogger.yaml (hunk 2 above)
Comment (ccordi) [reply test/config/example1.fabriclogger.yaml:1]: I slimmed it down a bit.  Let me know if you think the prometheus options should be removed.

File: config.schema.json (hunk 3 above)
Comment (ccordi) [reply config.schema.json:1]: This works now.

File: test/config/example1.fabriclogger.yaml (hunk 2 above)
Comment (ziegfried) [reply test/config/example1.fabriclogger.yaml:1]: Looking good 👍

**Review by atoulme — APPROVED** [review]

**Review by ziegfried — APPROVED** [review]

**Pull request merged**